src/visualization/convert.jl

```diff
@@ -73,6 +73,44 @@ function get_data_2d(center_level_0, length_level_0, leaf_cells, coordinates, le
 end
 
 
+# Extract data from a 1D DG solution and prepare it for visualization as a line plot.
+# This returns a tuple with
+# - x coordinates
+# - nodal 1D data
+#
+# Note: This is a low-level function that is not considered as part of Trixi's interface and may
+#       thus be changed in future releases.
+function get_data_1d(original_nodes, unstructured_data, nvisnodes)
+  # Get the dimensions of u; where n_vars is the amount of variables, a the amount of entries in each element and n_nodes the amount of elements.
+  n_nodes, n_elements, n_vars = size(unstructured_data)
+
+  # Set the amount of nodes visualized according to nvisnodes.
+  if nvisnodes === nothing
+    max_nvisnodes = 2 * n_nodes
+  elseif nvisnodes == 0
+    max_nvisnodes = n_nodes
+  else
+    max_nvisnodes = nvisnodes
+  end
+
+  interpolated_nodes = Array{Float64,2}(undef, max_nvisnodes, n_elements)
+  interpolated_data = Array{Float64,2}(undef, max_nvisnodes*n_elements, n_vars)
+
+  # Iterate over all elements.
+  for i=1:n_elements
+    # Interpolate on an equidistant grid.
+    interpolated_nodes[:,i] = vcat(range(original_nodes[1,1,i], original_nodes[1,end,i], length = max_nvisnodes))
+
+    # Interpolate the data for each variable.
+    for j=1:n_vars
+      interpolated_data[(i-1)*max_nvisnodes+1:i*max_nvisnodes, j] = interpolate1d(original_nodes[1,:,i], unstructured_data[:,i,j], interpolated_nodes[:,i])
```

Comment:
I think it would be sufficient here to create the Vandermonde matrix once, i.e., from the original node locations *in reference space* (i.e., the nodes between -1 and 1) to the corresponding uniform node locations in [-1, 1]. Then you do not have to create one for each element and can basically omit the entire `interpolace1d` call.

Reply:
I feel this is easily done once the comment above is resolved.
The current implementation of the vandermonde matrix depends on the exact location of each element, which means I cannot do this right now. This could be done with polynomial_interpolation_matrix, since it is not element-dependend.

Reply:
Are you sure about this? I don't think the absolute position of the nodes should matter for interpolation, only the position relative to each other.

Reply:
With my version of the vandermonde matrix I think this wasn't possible.
Since I changed to polynomial_interpolation_matrix this is no longer a problem and I could solve this in 97690d4.